core/silkworm/types/block.cpp

```diff
@@ -69,6 +70,9 @@ namespace rlp {
             rlp_head.payload_length += kHashLength + 1;  // mix_hash
             rlp_head.payload_length += 8 + 1;            // nonce
         }
+        if (header.base_fee_per_gas) {
+            rlp_head.payload_length += length(*header.base_fee_per_gas);
+        }
```

Comment:
For better readability I'd prefer always to check the `.has_value()`  method (if you don't mind).
This makes clear this is an std::optional regardless the contained value